web/auth0.go

```diff
@@ -92,57 +167,80 @@ func InstallAuth0(
 		httpTransport.CloseIdleConnections()
 		return nil, err
 	}
-	state.auth0HTTPTransport = &httpTransport
+	state.httpTransport = &httpTransport
 
 	state.authConfig = oauth2.Config{
 		ClientID:     config.ClientID,
 		ClientSecret: config.Secret,
-		RedirectURL:  config.BaseURL + "/callback",
+		RedirectURL:  config.BaseURL + redirectUrl,
 		Endpoint:     p.Endpoint(),
 		Scopes:       []string{oidc.ScopeOpenID, "profile", "email"},
 	}
 
-	mux.Handle(pat.New("/callback"), &callbackHandler{state, logger})
-	mux.Handle(pat.New("/login"), &loginHandler{state, logger})
-	mux.Handle(pat.New("/logout"), &logoutHandler{state, logger})
+	return state, nil
+}
 
-	if state.config.EnableTest {
-		mux.Handle(pat.New("/token-callback"), &tokenCallbackHandler{state, logger})
-	}
+func installAuthProviderRoutes(
+	mux *goji.Mux,
+	authProvider *AuthProvider,
+	providerLogoutURL string,
+	redirectURL string,
+	redirectStateCookieName string,
+	redirectStateCookieMaxAge time.Duration,
+	logger golog.Logger) {
+	mux.Handle(pat.New("/login"), &loginHandler{
+		authProvider,
+		logger,
+		redirectStateCookieName,
+		redirectStateCookieMaxAge})
+	mux.Handle(pat.New(redirectURL), &callbackHandler{
+		authProvider,
+		logger,
+		redirectStateCookieName})
+	mux.Handle(pat.New("/logout"), &logoutHandler{
+		authProvider,
+		logger,
+		providerLogoutURL,
+	})
+	mux.Handle(pat.New("/token"), &tokenHandler{
+		authProvider,
+		logger,
+		redirectStateCookieName,
+		redirectStateCookieMaxAge})
 
-	return state, nil
+	if authProvider.config.EnableTest {
+		mux.Handle(pat.New("/token-callback"), &tokenCallbackHandler{authProvider, logger})
+	}
 }
 
 type callbackHandler struct {
-	state  *auth0State
-	logger golog.Logger
+	state                   *AuthProvider
+	logger                  golog.Logger
+	redirectStateCookieName string
 }
 
-const (
-	auth0RedirectStateCookieName   = "auth0_redirect_state"
-	auth0RedirectStateCookieMaxAge = time.Minute * 10
-)
-
 func (h *callbackHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	ctx, cancel := context.WithTimeout(r.Context(), 10*time.Second)
 	defer cancel()
 
 	ctx, span := trace.StartSpan(ctx, r.URL.Path)
 	defer span.End()
 
-	stateCookie, err := r.Cookie(auth0RedirectStateCookieName)
+	stateCookie, err := r.Cookie(h.redirectStateCookieName)
 	if HandleError(w, err, h.logger, "getting redirect cookie") {
 		return
 	}
-	http.SetCookie(w, &http.Cookie{
-		Name:     auth0RedirectStateCookieName,
-		Value:    "",
-		Path:     "/",
-		MaxAge:   -1,
-		Secure:   r.TLS != nil,
-		SameSite: http.SameSiteLaxMode,
-		HttpOnly: true,
-	})
+
+	// http.SetCookie(w, &http.Cookie{
+	// 	Name:     h.redirectStateCookieName,
+	// 	Value:    "",
+	// 	Path:     "/",
+	// 	MaxAge:   -1,
+	// 	Secure:   r.TLS != nil,
+	// 	SameSite: http.SameSiteLaxMode,
+	// 	HttpOnly: true,
+	// })
```

Comment:
[q] Devin, so this should be turned off for FusionAuth only? Or for both Auth0 and FusionAuth?